### PR TITLE
copy input in send buffer to stop data loss, add ulid and caller to log shipper logs

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: pkg/log/logshipper/logshipper.go [hunk 1]
```diff
@@ -39,7 +40,7 @@ func New(k types.Knapsack, baseLogger log.Logger) *LogShipper {
 	}
 
 	sendBuffer := sendbuffer.New(sender, sendbuffer.WithSendInterval(sendInterval))
-	shippingLogger := log.NewJSONLogger(sendBuffer)
+	shippingLogger := log.With(log.NewJSONLogger(sendBuffer), "caller", log.Caller(6), "session_ulid", ulid.New())
```
Comment (James-Pickett) [inline pkg/log/logshipper/logshipper.go:43]: wonder if we should send the querier to get serial # or some kind of machine ID

Comment (directionless) [reply pkg/log/logshipper/logshipper.go:43]: I'd kick that forward. I think we want it, but I'm not sure where it'll make the most sense to add

Comment (directionless) [reply pkg/log/logshipper/logshipper.go:43]: What's the ulid for?

**Review by directionless — DISMISSED** [review]
Approved, though I don't know what the ulid is for.

File: pkg/log/logshipper/logshipper.go (hunk 1 above)
Comment (James-Pickett) [reply pkg/log/logshipper/logshipper.go:43]: The ulid is just uniquely identify the session, I think while the PID is sufficient for local logs to debug.json. We will have overlap when we aggregating 10,000s of logs.

**Review by directionless — APPROVED** [review]

**Pull request merged**